### ISSUE TITLE
[Google] Hide reference data from ingestion DURING scoring program

### ISSUE THREAD

**Issue by ckcollab** [codalab/codalab-competitions#2391]
"we could move the reference data outside normal access and give access to the scoring program Docker container. Like how we have hidden reference data for ingestion programs during prediction already. Could use the same `$hidden` variable name."

**Comment by ckcollab** [comment]
Blocked on `ContentCategory` problem

**Comment by zhengying-liu** [comment]
I added a $hidden argument to the metadata of scoring program to have:
```
command: python $program/score.py $hidden $input $output
```
But $hidden is replaced by an empty directory:
```
sys.argv =  ['/tmp/codalab/tmpEAx7ES/run/program/score.py', '', '/tmp/codalab/tmpEAx7ES/run/input', '/tmp/codalab/tmpEAx7ES/run/output']
```

The competition bundle used is:
[AutoDL_competition_bundle_2018-09-13-12-01-00.zip](https://github.com/codalab/codalab-competitions/files/2378813/AutoDL_competition_bundle_2018-09-13-12-01-00.zip)

**Comment by zhengying-liu** [comment]
New issue encountered:

For an originally working competition (http://134.158.74.63/competitions/1), when I made a new submission, it gives following error:
<img width="1184" alt="screen shot 2018-09-14 at 11 10 02" src="https://user-images.githubusercontent.com/20074659/45541338-1f109200-b80f-11e8-9991-195d05b8d2e7.png">

The type of the run seems to have changed? Is it not a scoring run anymore?

**Comment by ckcollab** [comment]
That data should be in `$hidden` now, available only to scoring program.

**Comment by zhengying-liu** [comment]
Ok, I would like to test it. But now, weirdly, ingestion program and scoring program seem to run forever (you can see the duration on the picture above) while I put a time budget of 60 seconds. 

Before, when I had some similar error, the run would be terminated immediately and I can check the log to find where the problem is. But now it's very difficult to do trouble shooting...

**Comment by zhengying-liu** [comment]
Ok, now $hidden is indeed replaced by a secret directory outside the run. We see in following Scoring Output that there is indeed no more input/res/ and $hidden is outside the run:
[scoring-stdout.txt](https://github.com/codalab/codalab-competitions/files/2384215/scoring-stdout.txt)

But another problem comes: the data used for training (in ingestion) was with reference data in input/res. Now since the reference data directory is no more accessible to ingestion program, where should I put these training data?

In other words, where can I find my **input_data** as in my YAML file:
<img width="422" alt="screen shot 2018-09-14 at 18 59 55" src="https://user-images.githubusercontent.com/20074659/45564379-8864c500-b850-11e8-9339-a4c7c9dc3b51.png">
